src/codc/encoder.py

```diff
@@ -50,20 +50,26 @@ def infer_column_type(arr, override_type):
         # Infer the column type from the data, if no column type given
 
         if dtype is None:
-            if arr.dtype in ("uint64", "int64"):
+            # Note that "uint64" is not supported
+            if arr.dtype in ("uint8", "int8", "uint16", "int16", "uint32", "int32", "int64"):
                 dtype = INTEGER
-            elif arr.dtype == "float64":
+            elif arr.dtype in ["float32", "float64"]:
                 if not data.isnull().all() and all(pandas.isnull(v) or float(v).is_integer() for v in arr):
                     dtype = INTEGER
-                else:
+                elif arr.dtype == "float32":
+                    dtype = REAL
+                elif arr.dtype == "float64":
                     dtype = DOUBLE
+
             if arr.dtype == "object" or pandas.api.types.is_string_dtype(arr):
                 if not arr.isnull().all() and all(s is None or isinstance(s, str) for s in arr):
                     dtype = STRING
                 elif arr.isnull().all():
                     dtype = INTEGER
 
         # With an inferred, or supplied column type, massage the data into a form that can be encoded
+        if dtype == INTEGER:
+            return_arr = arr.fillna(value=missing_integer).astype("int64")
```

Comment:
Can you confirm that this performs no additional work if the dtype is already int64.

IT looks to me like this should only be being called if the dtype is either floating point (--> filling with missing values), or 32 bit input. Might need to have slightly more careful guards in the if statement.

Reply:
I've added an extra check so that it only performs this step if the dtype is one of ("uint8", "int8", "uint16", "int16", "uint32", "int32"). The previous code also had the bug that if arr.dtype was float then line 72 would do work for no reason.